packages/wallets/src/evm/connectors/smart-wallet/index.ts

```diff
@@ -54,6 +54,7 @@ export class SmartWalletConnector extends Connector<SmartWalletConnectionArgs> {
       this.config.paymasterUrl ||
       `https://${this.chainId}.bundler.thirdweb.com/v2`;
     const entryPointAddress = config.entryPointAddress || ENTRYPOINT_ADDRESS;
+    const deployOnSign = config.deployOnSign || true;
```

Comment:
You have to use ?? Here, or explicitly check for undefined.

Otherwise if I set it to false as a user you still default me to true (I know Js is annoying)

Reply:
lol, fixed in https://github.com/thirdweb-dev/js/pull/2222/commits/5271006386959c15a5b8635b32c212e1e29dba2a